File: sol/2965/2965.cpp
struct Solution {
    vector<int> findMissingAndRepeatedValues(vector<vector<int>>& grid) {
        int n = grid.size();
        int total = 0, totalSq = 0, counter = 1;
        for (auto &row : grid){
            for (int x : row){
                total += x - counter;
                totalSq += x * x - counter * counter;
                ++counter;
            }
        }
        int c = totalSq / total;
        int a = (c + total) / 2;
        return {a, c - a};
    }
};

File: sol/2965/2965.py
class Solution:
    def findMissingAndRepeatedValues(self, grid: List[List[int]]) -> List[int]:
        n = len(grid)
        count = [0] * (n * n)
        for row in grid:
            for x in row:
                count[x - 1] += 1
        res = [0, 0]
        for i, v in enumerate(count, 1):
            if v == 0:
                res[1] = i
            elif v == 2:
                res[0] = i
        return res
